File: May 2022/May_01.java
class Solution {
    public boolean backspaceCompare(String s, String t) {
        Stack<Character> st1 = new Stack<>(), st2 = new Stack<>();
        for(char ch : s.toCharArray()){
            if(ch == '#'){
                if(!st1.isEmpty())
                    st1.pop();
            } else
                st1.push(ch);
        }
        for(char ch : t.toCharArray()){
            if(ch == '#'){
                if(!st2.isEmpty())
                    st2.pop();
            } else
                st2.push(ch);
        }
        
        while(!st1.isEmpty() && !st2.isEmpty())
            if(st1.pop() != st2.pop())
                return false;
        
        return st1.isEmpty() && st2.isEmpty();
    }
}

File: May 2022/May_01.cpp
class Solution {
public:
    bool backspaceCompare(string s, string t) {
        stack<char> st1, st2;
        
        for(auto &c : s){
            if(c == '#'){
                if(!st1.empty())
                    st1.pop();
            } else
                st1.push(c);
        }
        
        for(auto &c : t){
            if(c == '#'){
                if(!st2.empty())
                    st2.pop();
            } else
                st2.push(c);
        }
        
        while(!st1.empty() && !st2.empty()){
            if(st1.top() != st2.top())
                return false;
            st1.pop(); st2.pop();
        }
        
        return st1.empty() && st2.empty();
    }
};
